src/librbd/journal/Types.cc

```diff
@@ -71,16 +71,19 @@ class DumpVisitor : public boost::static_visitor<void> {
 void AioDiscardEvent::encode(bufferlist& bl) const {
   ::encode(offset, bl);
   ::encode(length, bl);
+  ::encode(force_discard, bl);
```

Comment:
You need to increment the encode/decode versions when changing structs.

[1] https://github.com/ceph/ceph/blob/master/src/librbd/journal/Types.cc#L321